src/main/java/org/reactome/addlinks/fileprocessors/ensembl/EnsemblBioMartFileProcessor.java

```diff
@@ -0,0 +1,167 @@
+package org.reactome.addlinks.fileprocessors.ensembl;
+
+import org.reactome.addlinks.EnsemblBioMartUtil;
+import org.reactome.addlinks.fileprocessors.FileProcessor;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Stream;
+
+public class EnsemblBioMartFileProcessor extends FileProcessor<Map<String, List<String>>> {
+
+    public EnsemblBioMartFileProcessor()
+    {
+        super(null);
+    }
+
+    public EnsemblBioMartFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    /**
+     * This class builds a large, rather unwieldy two-layer mapping structure of identifiers corresponding to a specific species.
+     * The first level has 4 types of keys for each species. These keys are 'uniprotToProteins', 'proteinToGenes',
+     * 'proteinToTranscripts', and 'transcriptToMicroarrays'. These describe the type of mapping that this
+     * key is to. For example, the key 'hsapiens_uniprotToProteins' would access a Map of Human UniProt protein identifiers
+     * to a List of Ensembl protein identifiers. Transcript denotes Ensembl transcript identifiers, Gene is Ensembl gene identifiers
+     * and Microarray denotes microarray probe identifiers.
+     * @return Map<String, Map<String, List<String>>>> mappings, the two-layer mapping structure of species-(identifier-[identifiers]).
+     * @throws IOException Can be thrown if file does not exist.
+     */
+
+    private static final int geneIdentifierIndex = 0;
+    private static final int transcriptIdentifierIndex = 1;
+    private static final int proteinIdentifierIndex = 2;
+    private static final int identifierIndex = 3;
+    @Override
+    public Map<String, Map<String, List<String>>> getIdMappingsFromFile() {
+
+        // Below is a sample of the structure of the final mappings object. Each species has up to 4 different mappings associated with it.
+        /*
```

Comment:
I know it's a bit long, but it might be worth including the sample of the mapping structure in the JavaDoc of the method before the annotations.